DEV_GUIDE.md

```diff
@@ -0,0 +1,163 @@
+
+###Content creation
+
+Scala Exercises content is organized in the following way:
+
+```
+Scala Exercises Core
+       |
+       | imports
+       |
+   Libraries
+       |
+       | contains
+       |
+    Sections
+       |
+       | contains
+       |
+    Exercises
+```
+
+####Creating a new content library
+
+Each content library is an independent SBT module working as a library, which should be published as an artifact for it to be included in Scala Exercises. From now on, we'll consider you've already set up a new project for your exercises library, and we'll focus on the specifics on how to layout the content itself.
+
+First of all you need to set up the library by creating a new Scala `object` that contains basic details about it. This `object` must subclass from `org.scalaexercises.definitions.Library`. Here you can find a template for this object:
+
+```scala
+package package_name (1)
+
+import org.scalaexercises.definitions._
+
+/** This line should describe your library shortly, and will appear as its description in the main Scala Exercises website. (2)
+  *
+  * @param name library_name (3)
+  */
+object MyLibrary extends Library {
+  override def owner = "owner" (4)
+  override def repository = "github_repository" (5)
+
+  override def color = Some("#HEX_COLOR") (6)
+
+  (7)
+  override def sections = List(
+    SectionA,
+    SectionB,
+    ...
+    SectionN
+  )
+}
+```
+
+Let's dig into each one of these points:
+
+* (1) A package name for your exercises library. 
+* (2) Inside the first ScalaDoc block, we define the description of the library as it'll appear in the list of the available libraries.
+* (3) Also, this ScalaDoc block defines the title of the library. It'll also be used as the path suffix for its URL. If you need to use spaces, use an underscore symbol (i.e.: "STD_lib")
+* (4) Owner username or organization in GitHub.
+* (5) Name of the repository that will contain the library.
+* (6) Color (in hexadecimal code format) that will identify the library in the website.
+* (7) A list that contains every section you want to include in this library. Every item defined here is a Scala object containing the exercises per section.
+
+####Creating a new section
+
+As previously mentioned, exercises are organized in sections that will be based on a Scala `object`. A good practice would be to have each section in a separate Scala file. So basically, the basic flow to create new exercises would be:
+
+1. Define the section `object` (we'll go through this process in a moment).
+2. Include a reference to this object in the `sections` list in the library `object` described in the previous section.
+
+How does a section look like?
+
+```scala
+/** @param name section_title (1)
+*/
+object MySection extends FlatSpec with Matchers with org.scalaexercises.definitions.Section (2) {
```

Comment:
This is missing the scalatest imports. Also no need to fully qualify the Section import if it's provided above